MBBSEmu/Program.cs

```diff
@@ -96,6 +118,18 @@ static void Main(string[] args)
                     }
                 }
 
+                if (sSettingsFile == null)
```

Comment:
@enusbaum Question about this `if` block and how C# default parameters work.

Can all this be condensed down to simply `ServiceResolver.Create(sSettingsFile)`?

I'm hoping that works, but my fear is that there's some subtle difference between the programmer explicitly setting `null`, vs. the variable implicitly being `null`, and how the default parameter in the `Create` function behaves.

Please tell me I'm overthinking this and we can delete some code here...

Reply:
You could define default file name in this class as a constant and do something like `ServiceResolver.Create(sSettingsFile ?? DefaultFile)` (?? being an if-null-then) :)

Reply:
Thanks - in learning more about `??` I also learned about `??=`. Ended up using that as IMO it removed some subtleties hiding in the syntax.